pybandits/offline_policy_evaluator.py

```
@@ -617,38 +601,62 @@ def _check_argument_required_by_estimators(cls, argument: str, ope_estimators: L
 
-    def _extract_batches(self):
+    def _extract_batches(self, logged_data: pd.DataFrame) -> Tuple[Dict[str, Any], Dict[str, Any]]:
         """
         Create training and test sets in dictionary form.
 
         """
-        logged_data = self.logged_data.sort_values(by=self.batch_feature)
+        self._validate_logged_data(logged_data)
+        logged_data = logged_data.sort_values(by=self.batch_feature)
         unique_batch = logged_data[self.batch_feature].unique()
         split_batch = unique_batch[int(floor(len(unique_batch) * self.split_prop))]
```

Ruff (0.15.4)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)